internal/database/external_services.go

```diff
@@ -1088,41 +1111,123 @@ func (e *externalServiceStore) GetByID(ctx context.Context, id int64) (*types.Ex
 	return ess[0], nil
 }
 
-func (e *externalServiceStore) GetSyncJobs(ctx context.Context) ([]*types.ExternalServiceSyncJob, error) {
-	q := sqlf.Sprintf(`SELECT id, state, failure_message, started_at, finished_at, process_after, num_resets, external_service_id, num_failures
-FROM external_service_sync_jobs ORDER BY started_at desc
-`)
+const getSyncJobsQueryFmtstr = `
+SELECT
+	id,
+	state,
+	failure_message,
+	queued_at,
+	started_at,
+	finished_at,
+	process_after,
+	num_resets,
+	external_service_id,
+	num_failures
+FROM
+	external_service_sync_jobs
+WHERE %s
+ORDER BY
+	started_at DESC
+%s
+`
+
+func (e *externalServiceStore) GetSyncJobs(ctx context.Context, opts ExternalServicesGetSyncJobsOptions) ([]*types.ExternalServiceSyncJob, error) {
+	var preds []*sqlf.Query
+
+	if opts.ExternalServiceID != 0 {
+		preds = append(preds, sqlf.Sprintf("external_service_id = %s", opts.ExternalServiceID))
+	}
+
+	if len(preds) == 0 {
+		preds = append(preds, sqlf.Sprintf("TRUE"))
+	}
+
+	q := sqlf.Sprintf(getSyncJobsQueryFmtstr, sqlf.Join(preds, "AND"), opts.LimitOffset.SQL())
 
 	rows, err := e.Query(ctx, q)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		err = basestore.CloseRows(rows, err)
+	}()
 
-	var jobs []*types.ExternalServiceSyncJob
+	jobs := make([]*types.ExternalServiceSyncJob, 0)
 	for rows.Next() {
 		var job types.ExternalServiceSyncJob
-		if err := rows.Scan(
-			&job.ID,
-			&job.State,
-			&dbutil.NullString{S: &job.FailureMessage},
-			&dbutil.NullTime{Time: &job.StartedAt},
-			&dbutil.NullTime{Time: &job.FinishedAt},
-			&dbutil.NullTime{Time: &job.ProcessAfter},
-			&job.NumResets,
-			&dbutil.NullInt64{N: &job.ExternalServiceID},
-			&job.NumFailures,
-		); err != nil {
+		if err := scanExternalServiceSyncJob(rows, &job); err != nil {
 			return nil, errors.Wrap(err, "scanning external service job row")
 		}
 		jobs = append(jobs, &job)
 	}
-	if rows.Err() != nil {
-		return nil, errors.Wrap(err, "row scanning error")
-	}
 
 	return jobs, nil
 }
 
+const countSyncJobsQueryFmtstr = `
+SELECT
+	COUNT(1)
```

Comment:
```suggestion
	COUNT(*)
```

https://www.citusdata.com/blog/2016/10/12/count-performance/#dup_counts_exact

Passing over knowledge from Tomas 😁